src/main/webapp/js/instructorFeedbackResults.js

```diff
@@ -205,37 +205,79 @@ function updateStatsCheckBox() {
     $('input[id=statsShownCheckBox]').val($('#show-stats-checkbox').is(':checked'));
 }
 
-function toggleCollapse(e, pans) {
-    var expand = 'Expand';
-    var collapse = 'Collapse';
-    var panels = pans || $('div.panel-collapse');
-
-    if ($(e).html().trim().startsWith(expand)) {
-        var i = 0;
-        for (var idx = 0; idx < panels.length; idx++) {
-            if ($(panels[idx]).attr('class').indexOf('in') === -1) {
-                setTimeout(showSingleCollapse, 50 * i, panels[idx]);
-                i++;
+/**
+ * Expand each panel. If the panel data is not loaded, load it by Ajax first before expansion.
+ */
+function expand(panels, needToCheckForAjaxLoading) {
+    var i = 0;
+    for (var idx = 0; idx < panels.length; idx++) {
+        if (needToCheckForAjaxLoading) {
```

Comment:
Unfortunately we cannot do it here. The else block must be executed after the very long block because the very long block might **continue** the for loop.  the code following the long block is dependent on the long block.